src/engine/strat_engine/backstore/util.rs

```diff
@@ -36,3 +39,44 @@ pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
 
     result
 }
+
+// Probes a block device using the libblkid library and returns Some(None) if the device is empty or
+// Some(String) with the String having similar content as the output of `blkid -p <device>.
+pub fn device_signature(path: &Path) -> Result<Option<String>, blkid::BlkidError> {
+
+    let d = blkid::BlkId::new(path)?;
+
+    d.enable_partitions()?
+        .enable_superblocks()?
+        .set_superblock_flags(blkid_sys::BLKID_SUBLKS_USAGE | blkid_sys::BLKID_SUBLKS_TYPE |
+                              blkid_sys::BLKID_SUBLKS_BADCSUM |
+                              blkid_sys::BLKID_SUBLKS_UUID)?;
+
+    // There is potential for the probe to fail if the device is in use/locked, lets loop a bit
```

Comment:
Why do we loop to help out blkid and not in lots of other places where there could also be some similar in use problem? Is there some special justfication for blkid?

Reply:
Utilizing https://github.com/storaged-project/libblockdev/blob/master/src/plugins/fs/generic.c#L238 as a template on `liblkid`.  In general I think any code that accesses a device during udev handing event may run into device contention, so yeah maybe this should be considered in other cases too.